packages/core/src/types.ts

```diff
@@ -48,3 +50,26 @@ export interface SuggestionInput {
   arboristValues: IArboristNode[];
   latestPackages?: IDependencyMap;
 }
+
+export interface Plugin {
+  name: string;
+  version: string;
+  // Plugin data is serialized to the report and available during render time
+  pluginData: any;
+  suggestions: Suggestion[];
+}
+
+export interface PluginServer extends Plugin {
+  getSuggestions?(suggestionInput: SuggestionInput): Promise<Suggestion[]>;
+}
+
+export interface PluginBrowser extends Plugin {
+  // View rendered as a block on the report page
+  reportView?: React.FC;
+  // Renders as a full page experience
+  pluginPageView?: React.FC;
+  // Renders on the package page
+  packageView?: React.FC;
+  // Renders the card view for a given suggestion
+  cardView?: React.FC<{ suggestions: Suggestion[] }>;
+}
```

Comment:
Same suggestion for naming to use `ServerPlugin` and `BrowserPlugin`

Reply:
will do.